tools/README.md

```diff
@@ -0,0 +1,154 @@
+- `createE2eConfig(projectKey, { projectRoot, ...options })`
+
+`projectKey` is used for cache and coverage directories.
+
+### Defaults
+
+Common to all kinds:
+
+- `reporters: ['basic']`, `globals: true`, `environment: 'node'`
+- `alias: tsconfigPathAliases()`
+- `pool: 'threads'` with `singleThread: true`
+- Cache directories resolved from `projectRoot` (absolute paths)
+
+Coverage:
+
+- Unit/Int: enabled by default, reports to `<projectRoot>/coverage/<project>/<kind>-tests`
+- E2E: disabled by default
+- Default exclude: `['mocks/**', '**/types.ts']`
+
+Global setup:
+
+- Unit/Int: `['<projectRoot>/global-setup.ts']` by default
+- E2E: none by default (set per-suite if needed)
+
+Include patterns:
+
+- Unit: `src/**/*.unit.test.*`
+- Int: `src/**/*.int.test.*`
+- E2E: `tests/**/*.e2e.test.*`
+
+### setupFiles strategy
+
+Baseline `setupFiles` are injected automatically by kind:
+
+- Unit baseline: `console.mock.ts`, `reset.mocks.ts`
+- Int baseline: `console.mock.ts`, `reset.mocks.ts`
+- E2E baseline: `reset.mocks.ts`
+
+Extend with additional files using `options.setupFiles` — they append after the baseline (paths are project-root-relative):
+
+```ts
+export default createUnitConfig('core', {
+  projectRoot: new URL('../../', import.meta.url),
+  setupFiles: ['testing/test-setup/src/lib/cliui.mock.ts'],
+});
+```
+
+Replace entirely using `overrideSetupFiles: true` (paths are project-root-relative):
+
+```ts
+export default createUnitConfig('core', {
+  projectRoot: new URL('../../', import.meta.url),
+  overrideSetupFiles: true,
+  setupFiles: ['testing/test-setup/src/lib/cliui.mock.ts', 'testing/test-setup/src/lib/fs.mock.ts'],
+});
+```
+
+### Using presets directly
+
+`vitest-setup-presets.ts` exposes grouped arrays you can compose if needed:
+
+```ts
+import { setupPresets } from '../../tools/vitest-setup-presets.js';
+
+export default createIntConfig('core', {
+  projectRoot: new URL('../../', import.meta.url),
+  setupFiles: [...setupPresets.int.portalClient],
+});
+```
+
+Preset keys:
+
+- `setupPresets.unit.{base,cliui,fs,git,portalClient,matchersCore,matcherPath}`
+- `setupPresets.int.{base,cliui,fs,git,portalClient,matcherPath,chromePath}`
+- `setupPresets.e2e.{base}`
+
+### Options reference
+
+`CreateVitestConfigOptions` (required + optional):
+
+- `projectKey` (string): coverage/cache naming.
+- `kind` ('unit' | 'int' | 'e2e'): test kind.
+- `projectRoot` (string | URL): absolute root for all paths.
+- `include?: string[]`: override default include globs.
+- `setupFiles?: string[]`: extra setup files (appended to baseline; project-root-relative).
+- `overrideSetupFiles?: boolean`: skip baseline and use only provided list.
+- `globalSetup?: string[]`: override default global setup (project-root-relative).
+- `coverage?: { enabled?, exclude?, reportsSubdir? }`
+- `testTimeout?: number`: e.g., for E2E.
+- `typecheckInclude?: string[]`: include patterns for Vitest typecheck.
+- `cacheKey?: string`: custom cache dir suffix.
```

Comment:
The best DX is given if the user does no need to lear another type. 

I would suggest try to keep the types as close as possible to existing virest types. 

e.g. :  `createIntConfig({/*vitestTestConfig*/}, { projectKey: 'unit', /*other custom options we need that are not already inside vitest types */})`